### DISCUSSION TITLE
Unix Timestamp to Date-Time

### DISCUSSION THREAD

**Discussion by redboyhun** [pspete/psPAS#466] [Ideas]
It would be nice if the timestamps would be converted to normal date time.
eg. in Get-PASComponentDetails.ps1

$result | Select-Object -ExpandProperty ComponentsDetails | select ComponentIP,ComponentUserName,ComponentVersion,IsLoggedOn,@{Name="LastLogonDate";Expression={((Get-Date 01.01.1970)+([System.TimeSpan]::fromseconds($_.LastLogonDate))).ToString("yyyy-MM-dd HH:mm:ss")}}

**Comment by pspete** [comment]
Datetimes returned from the API as UnixTime are formatted to be readable - this is all handled in the [`*formats.ps1xml`] (https://github.com/pspete/psPAS/tree/master/psPAS/xml) files in the module.
Am not aware of methods to do this for nested properties though... so only implemented for properties at the root level of the output object.

**Comment by JohnnyLeuthard** [comment]
My way around it is to just create a custom column a use the select command to convert the time
Something like this with $Data being a variable containing the contents

$CustomCoulum = @{'Name'='FriendlyTime';E={(get-date 1/1/70).addseconds($_.UNIXTIMEHEADERHERE) }}
$Data | Select *,$CustomCoulum
